toucan_connectors/http_api/authentication_configs.py

```diff
@@ -246,14 +249,17 @@ def _get_access_token(self) -> str:
         oauth_token = self._secrets_keeper.load(self.auth_flow_id)
         if oauth_token is None:
             raise ValueError("No oauth token found. Please refresh your oauth2 access.")
-        if datetime.fromtimestamp(oauth_token.expires_at) < datetime.utcnow():
-            client = self._init_oauth_client()
-            new_token = client.refresh_token(self.token_url, refresh_token=oauth_token.refresh_token)
-            oauth_token = self._secrets_keeper.save(
-                self.auth_flow_id,
-                # refresh call doesn't always contain the refresh_token
-                new_token | {"refresh_token": oauth_token.refresh_token},
-            )
+        if oauth_token.refresh_token:
+            # If refresh token exists, we want to verify if the access_token is still valid
+            # or if it must be refreshed.
+            if datetime.fromtimestamp(oauth_token.expires_at) < datetime.utcnow():
```

Comment:
```suggestion
            if datetime.fromtimestamp(oauth_token.expires_at) <= datetime.utcnow():
```

don't we want a little safety margin here ?  Like 30s ?

Reply:
Earlier or later? 

I'd say a maybe-unnecessary refresh is not a problem, but returning a token really close to expiration may lead to issues down the road (in which case, I second the safety margin).

Reply:
[649ff5a](https://github.com/ToucanToco/toucan-connectors/pull/1891/commits/649ff5a55939f8e44fad2ae3813a6567f7ed9c8f)